app/controllers/outputs_controller.rb

```diff
@@ -22,6 +22,19 @@ def create
     end
   end
 
+  # Override render and redirect_to so that we sign the invited user out
+  # before we render or redirect anywhere.
+  def render(*args)
```

Comment:
As far as I could see, after_action takes place *after* whatever your action does, which includes rendering or redirecting, so the user would see a page where they appear to be signed in, or they could in theory be redirected to somewhere they're not allowed to be. (A shame, because this does feel a bit hacky I know)